PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseStorageAPIsWithLengthParameter.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class UseStorageAPIsWithLengthParameter extends LocalInspectionTool {
+    @Nls
+    @NotNull
+    @Override
+    public String getDisplayName() {
+        return "Use Storage APIs With Length Parameter";
```

Comment:
Could use: ""Azure Storage upload API without length parameter detected"
Lets add more description to the rule too:
"This approach can lead to OutOfMemoryErrors, especially when dealing with large files or high-volume uploads"

Reply:
added this to the README as an explanation with a solution